### ISSUE TITLE
How to deploy Indy image to Kubernetes

### ISSUE THREAD

**Issue by tadayosi** [Commonjava/commonjava-images#82]
Hi,

After #80, now I want to deploy Indy to a K8s cluster so that it can be used as an alternative to Nexus Repo Manager. I've come up with my own primitive settings (see below) but I'd like to know your recommended settings with optionally more advanced features, such as externalised etc and data volumes.

Could you please add some documentation and/or examples on how to deploy Indy to K8s/OpenShift?

Thank you!

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: indy
  namespace: indy
spec:
  selector:
    matchLabels:
      app: indy
  template:
    metadata:
      labels:
        app: indy
    spec:
      containers:
        - name: indy
          image: quay.io/factory2/indy:2.7.2
          ports:
            - name: 8080-tcp
              containerPort: 8080
            - name: 8081-tcp
              containerPort: 8081
          livenessProbe:
            httpGet:
              path: /api/stats/version-info
              port: 8080
            initialDelaySeconds: 30 
            periodSeconds: 3
          readinessProbe:
            httpGet:
              path: /api/stats/version-info
              port: 8080
            initialDelaySeconds: 30
            periodSeconds: 3
          volumeMounts:
            - name: indy-etc
              mountPath: /opt/indy/etc/indy
      initContainers:
        - name: init-etc
          image: busybox
          command: ["sh", "-c"]
          args:
            - cd /tmp && wget -O etc.tar.gz https://github.com/Commonjava/indy/releases/download/indy-parent-2.7.2/indy-launcher-2.7.2-etc.tar.gz && tar -xf etc.tar.gz && mv etc/* /opt/indy/etc/indy/
          volumeMounts:
            - name: indy-etc
              mountPath: /opt/indy/etc/indy
      volumes:
        - name: indy-etc
          emptyDir: {}
```

**Comment by ligangty** [comment]
Hello @tadayosi I have two proposals:
1. Makes the indy configuration as configmap instead of a sidecar. You can create two configmaps for it: main.conf and logback.xml. I've merged a new pr with a merged main.conf in one file which can make it easily managed as configmap, please see new code.
2. If you want to persist all artifacts for your environment, please create a pvc to store all them. You can mount this pvc at /opt/indy/var/lib/indy (this can be conigured through "storage.dir" under [storage-default] section in main.conf).

**Comment by tadayosi** [comment]
Thanks @ligangty!  Actually initially I tried the configmap approach (see my setup below) but somehow it didn't succeed to deploy. The contents of the configmaps were created from https://github.com/Commonjava/indy/releases/download/indy-parent-2.7.2/indy-launcher-2.7.2-etc.tar.gz.

For the item 2, thans for your suggestion.

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: indy
  namespace: indy
spec:
  selector:
    matchLabels:
      app: indy
  template:
    metadata:
      labels:
        app: indy
    spec:
      containers:
        - name: indy
          image: quay.io/factory2/indy:2.7.2
          ports:
            - name: 8080-tcp
              containerPort: 8080
            - name: 8081-tcp
              containerPort: 8081
          livenessProbe:
            httpGet:
              path: /api/stats/version-info
              port: 8080
            initialDelaySeconds: 30
            periodSeconds: 3
          readinessProbe:
            httpGet:
              path: /api/stats/version-info
              port: 8080
            initialDelaySeconds: 30
            periodSeconds: 3
          volumeMounts:
            - name: indy-etc
              mountPath: /opt/indy/etc/indy/main.conf
              subPath: main.conf
            - name: indy-etc-confd
              mountPath: /opt/indy/etc/indy/conf.d
            - name: indy-etc-keycloak
              mountPath: /opt/indy/etc/indy/keycloak
            - name: indy-etc-logging
              mountPath: /opt/indy/etc/indy/logging
      volumes:
        - name: indy-etc
          configMap:
            name: indy-etc
        - name: indy-etc-confd
          configMap:
            name: indy-etc-confd
        - name: indy-etc-keycloak
          configMap:
            name: indy-etc-keycloak
        - name: indy-etc-logging
          configMap:
            name: indy-etc-logging
```

**Comment by ligangty** [comment]
@tadayosi Sorry for late reply. For the configmap stuff, do you have any error logs for tracking?
